### PR TITLE
fix: Colors in EmptySearchResultsView 

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we update the colors in the EmptySearchResultsView. In addition to that we get rid of the Color Scheme Variant from that view. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by ghuMustafa — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+
+final class LinkButton: DynamicFontButton {
+
+    init(title: String) {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift:24]: But "Like button" doesn't have a title. Why do we need it?

Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift:24]: You're right that was left from the initial implementation. If we need it in the future I will add it to the initialiser.

File: Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift [hunk 2]
```diff
@@ -76,29 +79,26 @@ final class EmptySearchResultsView: UIView {
...
-    init(variant: ColorSchemeVariant,
-         isSelfUserAdmin: Bool,
+    init(isSelfUserAdmin: Bool,
          isFederationEnabled: Bool) {
-        self.variant = variant
         self.isSelfUserAdmin = isSelfUserAdmin
         self.isFederationEnabled = isFederationEnabled
         stackView = UIStackView()
-        actionButton = InviteButton(variant: variant)
+        actionButton = LinkButton(title: "")
```
[the remark this answers is not in the document]
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift:99]: We don't, I removed it.

File: Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+
+final class LinkButton: DynamicFontButton {
+
+    init() {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift:24]: Maybe we can remove init()?

File: Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        super.init()
+
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setUpButtonStyle(title: String) {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift:33]:
```suggestion
    func setupButtonStyle(title: String) {
```

File: Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift [hunk 5]
```diff
@@ -76,29 +79,26 @@ final class EmptySearchResultsView: UIView {
         }
     }
 
-    private let variant: ColorSchemeVariant
     private let isSelfUserAdmin: Bool
     private let isFederationEnabled: Bool
 
     private let stackView: UIStackView
     private let iconView     = UIImageView()
-    private let statusLabel  = UILabel()
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift:85]: Why do you remove statusLabel?

File: Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        let attributes: [NSAttributedString.Key: Any] = [
+            .font: FontSpec.normalRegularFont.font!,
+            .foregroundColor: SemanticColors.Button.textUnderlineEnabled,
+            .underlineStyle: NSUnderlineStyle.single.rawValue
+        ]
+
+        let attributeString = NSMutableAttributedString(
+            string: title,
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift:41]: Like button doesn't have a title, why do we need it?

File: Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift [hunk 7]
```diff
@@ -49,14 +49,17 @@ protocol EmptySearchResultsViewDelegate: AnyObject {
...
     private var state: EmptySearchResultsViewState = .noUsers {
         didSet {
             iconView.image = icon
+            iconView.tintColor = iconColor
             statusLabel.text = self.text
 
             if let action = self.buttonAction {
                 actionButton.isHidden = false
-                actionButton.setTitle(action.title, for: .normal)
+                actionButton.setUpButtonStyle(title: action.title)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift:62]: I think setupStyle functions shouldn't contain the value. We use this function to define styles.

File: Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+
+final class LinkButton: DynamicFontButton {
+
+    func setupButtonStyle(title: String) {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Views/LinkButton.swift:24]: I would give the name setup(...) and keep the parameter `title`

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**